### DISCUSSION TITLE
Grid Item_Dropped signal only giving weakref for item?

### DISCUSSION THREAD

**Discussion by Karbonicc** [peter-kish/gloot#124] [Q&A] [answered]
Hi, I'm trying to use the item_dropped signal on the Grid inventory, however, the "item" field of the signal only gives a weak reference to the connect script, which doesn't allow me to get any information from the item itself. Am I doing something wrong?

For instance, using print(item) on the inventory_item_activated(item:var) signal gives something like "@Node@43:<Node#94774496325>"
while using print(item) on item_dropped(item:var, offset:var) only gets something like "<WeakRef#-9223371906797793584>"

**Comment by peter-kish** [comment, marked as answer]
You should still be able to access the `InventoryItem` via `WeakRef.get_ref` (bare in mind that it can return `null` - see the docs for [WeakRef](https://docs.godotengine.org/en/stable/classes/class_weakref.html)) . Something like:

```gdscript
var item_node: InventoryItem = item.get_ref()
if item_node == null:
    return
# do something with item_node
```

I used weak references here, since in cases where the item is dropped into another inventory, there's the possibility of it being "consumed" if it gets merged with another item. In that case the weak reference will contain `null`.

**Reply by Karbonicc** [reply]
Gotcha! Thanks so much for the quick response! Really great work on this addon, by the way!
